ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+WKNavigationDelegate.swift

```diff
@@ -1322,10 +1332,17 @@ extension BrowserViewController {
 
     tab?.externalAppAlertCounter += 1
 
-    return await withCheckedContinuation { continuation in
-      showExternalSchemeAlert(isSuppressActive: tab?.externalAppAlertCounter ?? 0 > 2) {
-        continuation.resume(with: .success($0))
+    return await withTaskCancellationHandler {
+      return await withCheckedContinuation { [weak tab] continuation in
+        tab?.externalAppPopupContinuation = continuation
+        showExternalSchemeAlert(isSuppressActive: tab?.externalAppAlertCounter ?? 0 > 2) {
+          tab?.externalAppPopupContinuation = nil
+          continuation.resume(with: .success($0))
+        }
       }
+    } onCancel: { [weak tab] in
```

Comment:
do you still have to nil `externalAppPopupContinuation` if you weak reference tab?

Reply:
Yes. The continuation needs to have its reference set to nil or it won't be freed until the tab dies.